router/transformer/transformer_test.go

```diff
@@ -0,0 +1,323 @@
+			},
+		},
+		"ctx_cancel_dest": {
+			name: "should throw timeout exception due to context getting cancelled immediately",
+			Expected: TcExpected{
+				StCode:      http.StatusInternalServerError,
+				Body:        `Post "%s/v0/destinations/ctx_cancel_dest/proxy": context canceled`,
+				ContentType: "text/plain; charset=utf-8",
+			},
+			Proxy: TcProxy{
+				StCode:   http.StatusOK,
+				Response: `{"output": {"status": 200, "message": "", "destinationResponse":"ctx_cancel_dest"}}`,
+			},
+			context: ProxyContextTc{
+				Cancel: true,
+			},
+			postParametersT: integrations.PostParametersT{
+				Type:          "REST",
+				URL:           "http://www.ctx_timeout_dest.domain.com",
+				RequestMethod: http.MethodPost,
+				QueryParams:   map[string]interface{}{},
+				Body: map[string]interface{}{
+					"JSON": map[string]interface{}{
+						"key_1": "val_1",
+						"key_2": "val_2",
+					},
+					"FORM":       map[string]interface{}{},
+					"JSON_ARRAY": map[string]interface{}{},
+					"XML":        map[string]interface{}{},
+				},
+				Files: map[string]interface{}{},
+			},
+		},
+		"not_found_dest": {
+			name: "should fail with not found error for not_found_dest",
+			Expected: TcExpected{
+				StCode:      http.StatusNotFound,
+				Body:        `post "%s/v0/destinations/not_found_dest/proxy" not found`,
+				ContentType: "text/plain; charset=utf-8",
+			},
+			Proxy: TcProxy{
+				StCode:   http.StatusNotFound,
+				Response: `Not Found`,
+			},
+			rtTimeout: 10 * time.Millisecond,
+			postParametersT: integrations.PostParametersT{
+				Type:          "REST",
+				URL:           "http://www.not_found_dest.domain.com",
+				RequestMethod: http.MethodPost,
+				QueryParams:   map[string]interface{}{},
+				Body: map[string]interface{}{
+					"JSON": map[string]interface{}{
+						"key_1": "val_1",
+						"key_2": "val_2",
+					},
+					"FORM":       map[string]interface{}{},
+					"JSON_ARRAY": map[string]interface{}{},
+					"XML":        map[string]interface{}{},
+				},
+				Files: map[string]interface{}{},
+			},
+		},
+	}
+)
+
+func Initialization() {
+	config.Load()
+	stats.Init()
+	stats.Setup()
+	rtTf.Init()
+	logger.Init()
+}
+
+func TestProxyRequest(t *testing.T) {
+	t.Setenv("RSERVER_HTTP_CLIENT_TIMEOUT", "1s")
+	Initialization()
+
+	srvMux := mux.NewRouter()
+	srvMux.HandleFunc("/v0/destinations/{destName}/proxy", proxyHandlerFunc)
+	srv := httptest.NewServer(srvMux)
+	defer srv.Close()
+
+	for destName, tc := range proxyResponseTcs {
+		// skip tests for the mentioned destinations
+		if destName == "not_found_dest" {
+			continue
+		}
+		t.Run(tc.name, func(t *testing.T) {
+			tr := rtTf.NewTransformer()
+			// Logic for executing test-cases not manipulating test-cases
+			if tc.rtTimeout.Milliseconds() > 0 {
+				tr.Setup(tc.rtTimeout)
+			} else {
+				// Just a default value
+				tr.Setup(2 * time.Millisecond)
+			}
+			// Logic to include context timing out
+			ctx := context.TODO()
```

Comment:
[minor] Is there a plan to provide a parent context? If not, TODO is not need:
```suggestion
			ctx := context.Background()
```

Reply:
I didn't completely understand what you meant here. Here's the change I guess you wanted me to make, kindly correct my understanding
```go
			parentctx := context.Background()
			var ctx context.Context
			var cancelFunc context.CancelFunc
			if tc.context.Timeout.Milliseconds() > 0 {
				ctx, cancelFunc = context.WithTimeout(parentctx, tc.context.Timeout)
				defer cancelFunc()
			} else if tc.context.Cancel {
				ctx, cancelFunc = context.WithCancel(parentctx)
				cancelFunc()
			} else {
				ctx = parentctx
			}
```

Reply:
@lvrach can you once see if this is the way we want to implement ?